base/system/services/rpcserver.c

```diff
@@ -1754,58 +1768,61 @@ RSetServiceStatus(
     /* Restore the previous service type */
     lpService->Status.dwServiceType = dwPreviousType;
 
-    /* Dereference a stopped service */
-    if ((lpServiceStatus->dwServiceType & SERVICE_WIN32) &&
-        (lpServiceStatus->dwCurrentState == SERVICE_STOPPED))
+    DPRINT("Service %S changed state %d to %d\n", lpService->lpServiceName, dwPreviousState, lpServiceStatus->dwCurrentState);
+
+    if (lpServiceStatus->dwCurrentState != SERVICE_STOPPED &&
```

Comment:
See also comment https://github.com/reactos/reactos/pull/1225/files#r245824492
(and same for the other if-case just below)

Reply:
I believe that this should be covered in another PR and I consider it out-of-scope for this one at this time. Thanks.